pkg/cache/external/scheduler_cache.go

```diff
@@ -132,6 +132,45 @@ func (cache *SchedulerCache) removeNode(node *v1.Node) error {
 	return nil
 }
 
+func (cache *SchedulerCache) AddPendingPodAllocation(podKey string, nodeID string) {
+	cache.lock.Lock()
+	defer cache.lock.Unlock()
+	delete(cache.inProgressAllocations, podKey)
+	cache.pendingAllocations[podKey] = nodeID
+}
+
+func (cache *SchedulerCache) RemovePendingPodAllocation(podKey string) {
+	cache.lock.Lock()
+	defer cache.lock.Unlock()
+	delete(cache.pendingAllocations, podKey)
+	delete(cache.inProgressAllocations, podKey)
+}
+
+func (cache *SchedulerCache) GetPendingPodAllocation(podKey string) (nodeID string, ok bool) {
+	cache.lock.RLock()
+	defer cache.lock.RUnlock()
+	res, ok := cache.pendingAllocations[podKey]
+	return res, ok
+}
+
+func (cache *SchedulerCache) GetInProgressPodAllocation(podKey string) (nodeID string, ok bool) {
+	cache.lock.RLock()
+	defer cache.lock.RUnlock()
+	res, ok := cache.inProgressAllocations[podKey]
+	return res, ok
+}
+
+func (cache *SchedulerCache) StartPodAllocation(podKey string, nodeID string) bool {
+	cache.lock.Lock()
+	defer cache.lock.Unlock()
+	expectedNodeID, ok := cache.pendingAllocations[podKey]
+	if ok && expectedNodeID == nodeID {
```

Comment:
Ok, from what I understand `expectedNodeID` comes from YK itself, and `nodeID` from the default scheduler. Questions:
* How often do these two mismatch?
* What happens if we return `false` here?
* Can YK and the def. scheduler end up repeatedly allocating to different nodes? 

Another thing is that do we even need to consider YK's allocation? Because in `task.postTaskAllocated()`, the original code path calls `KubeClient.Bind()` but in plugin mode, we delegate this operation as I can see.

Reply:
This code is used to verify that the requested node (by the default scheduler) is the one that YuniKorn decided to allocate the pod to. Returning `false` here will result in the Filter() call in the plugin returning `Unschedulable` to the default scheduler for this pod / node combination, and a subsequent Filter() call with a matching node will return `Success` instead. Effectively, we use this code to veto node allocations other than the node that YK has decided on.

The bind is deferred to the default scheduler logic once Filter() succeeds, which is why we don't handle the bind logic in postTaskAllocated().